hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpScheduledTransactionsSubcommand.java

```diff
@@ -0,0 +1,252 @@
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TreeMap;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/** Dump all scheduled transactions from a signed state file to a text file in a deterministic order  */
+public class DumpScheduledTransactionsSubcommand {
+    static void doit(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path scheduledTxsPath,
+            @NonNull final DumpStateCommand.EmitSummary emitSummary,
+            @NonNull final SignedStateCommand.Verbosity verbosity) {
+        new DumpScheduledTransactionsSubcommand(state, scheduledTxsPath, emitSummary, verbosity).doit();
+    }
+
+    @NonNull
+    final SignedStateHolder state;
+
+    @NonNull
+    final Path scheduledTxsPath;
+
+    @NonNull
+    final DumpStateCommand.EmitSummary emitSummary;
+
+    @NonNull
+    final SignedStateCommand.Verbosity verbosity;
+
+    DumpScheduledTransactionsSubcommand(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path scheduledTxsPath,
+            @NonNull final DumpStateCommand.EmitSummary emitSummary,
+            @NonNull final SignedStateCommand.Verbosity verbosity) {
+        requireNonNull(state, "state");
+        requireNonNull(scheduledTxsPath, "signedTxsPath");
+        requireNonNull(emitSummary, "emitSummary");
+        requireNonNull(verbosity, "verbosity");
+
+        this.state = state;
+        this.scheduledTxsPath = scheduledTxsPath;
+        this.emitSummary = emitSummary;
+        this.verbosity = verbosity;
+    }
+
+    void doit() {
+        final var scheduledTransactions = state.getScheduledTransactions().byId();
+        System.out.printf("=== %d scheduled transactions ===%n", scheduledTransactions.size());
+
+        final var allScheduledTxs = gatherScheduledTransactions(scheduledTransactions);
+
+        int reportSize;
+        try (@NonNull final var writer = new Writer(scheduledTxsPath)) {
+            if (emitSummary == DumpStateCommand.EmitSummary.YES) reportSummary(writer, allScheduledTxs);
+            reportOnScheduledTransactions(writer, allScheduledTxs);
+            reportSize = writer.getSize();
+        }
+
+        System.out.printf("=== scheduled transactions report is %d bytes%n", reportSize);
+    }
+
+    @SuppressWarnings(
+            "java:S6218") // "Equals/hashcode method should be overridden in records containing array fields" - this
+    record ScheduledTransaction(
+            long number,
+            @NonNull Key grpcAdminKey,
+            @NonNull Optional<JKey> adminKey,
+            @NonNull String memo,
+            boolean deleted,
+            boolean executed,
+            boolean calculatedWaitForExpiry,
+            boolean waitForExpiryProvided,
+            @Nullable EntityId payer,
+            EntityId schedulingAccount,
+            RichInstant schedulingTXValidStart,
```

Comment:
`@NonNull` for `schedulingAccount` and `schedulingTXValidStart` I presume?  Probably also `bodyBytes` ... `signatories` down at L119-122 as well?